src/main/java/it/gov/pagopa/admissibility/mapper/Initiative2InitiativeConfigMapper.java

```diff
@@ -23,7 +26,29 @@ public InitiativeConfig apply(Initiative2BuildDTO initiative) {
                 .beneficiaryInitiativeBudget(initiative.getGeneral().getBeneficiaryBudget())
                 .startDate(ObjectUtils.firstNonNull(initiative.getGeneral().getRankingStartDate(), initiative.getGeneral().getStartDate()))
                 .endDate(ObjectUtils.firstNonNull(initiative.getGeneral().getRankingEndDate(), initiative.getGeneral().getEndDate()))
-                .serviceId(initiative.getAdditionalInfo() != null? initiative.getAdditionalInfo().getServiceId() : null)
+                .serviceId(initiative.getAdditionalInfo() != null ? initiative.getAdditionalInfo().getServiceId() : null)
+                .rankingInitiative(initiative.getRankingInitiative())
+                .rankingFieldCodes(Boolean.TRUE.equals(initiative.getRankingInitiative()) ? retrieveRankingFieldCodes(initiative) : null)
                 .build();
     }
+
+    private List<String> retrieveRankingFieldCodes(Initiative2BuildDTO initiative) {
+        List<String> outListCode = null;
+
+        List<AnyOfInitiativeBeneficiaryRuleDTOSelfDeclarationCriteriaItems> selfDeclarationCriteria = initiative.getBeneficiaryRule()
+                .getSelfDeclarationCriteria();
+        if(selfDeclarationCriteria != null) {
+            List<String> codeTrue = selfDeclarationCriteria.stream()
+                    .filter(item -> item instanceof SelfCriteriaBoolDTO selfCriteriaBoolDTO && Boolean.TRUE.equals(selfCriteriaBoolDTO.getValue()))
+                    .map(selfCriteria -> ((SelfCriteriaBoolDTO) selfCriteria).getCode())
+                    .toList();
+
+            outListCode = initiative.getBeneficiaryRule().getAutomatedCriteria()
+                    .stream().map(AutomatedCriteriaDTO::getCode)
+                    .filter(codeTrue::contains)
```

Comment:
here you have as first filter by orderEnabled == true
next you have to map to retrieve just the code